src/util.ts

```diff
@@ -22,3 +22,24 @@ export function normalize<T = {}, U = Function>(
       (typeof optionsOrCallback === 'function' ? optionsOrCallback : cb)! as U;
   return {options, callback};
 }
+
+/**
+ * Return a Date object as a String, using the ISO8601 format:
+ * 20190308T220000Z
+ * @param date the Date object to convert.
+ * @internal
+ */
+export function dateToISOString(date: Date): string {
+  return date.toJSON()
+             .replace(/[:-]/g, '')  // strip :'s and -'s
+             .split('.')[0] +
+      'Z';  // remove milliseconds part
+}
+
+/**
+ * Flatten an object into an Array of arrays, [[key, value], ..].
+ * @internal
+ */
+export function flattenObject<T>(obj: {[key: string]: T}): Array<[string, T]> {
```

Comment:
isn't this pretty much the same as `Object.entries`?

Reply:
`Object.entries` is ES6+ as I've learned.. Can I use `lodash.entries` here?

Reply:
aghghghg - use whatever you like, then file an issue in the repo to clean it up once node 6 is EOL

Reply:
okay. I'll leave it as-is and add the issue after this is merged.